lib/elixir/lib/code/formatter.ex

```diff
@@ -1397,14 +1397,17 @@ defmodule Code.Formatter do
   defp interpolation_to_algebra([entry | entries], escape, state, acc, last) do
     {:"::", _, [{{:., _, [Kernel, :to_string]}, meta, [quoted]}, {:binary, _, _}]} = entry
     {doc, state} = block_to_algebra(quoted, line(meta), closing_line(meta), state)
-    doc = surround("\#{", doc, "}")
+    doc = interpolate_doc(doc)
     interpolation_to_algebra(entries, escape, state, concat(acc, doc), last)
   end
 
   defp interpolation_to_algebra([], _escape, state, acc, last) do
     {concat(acc, last), state}
   end
 
+  defp interpolate_doc({:doc_force, _t} = doc), do: surround("\#{", doc, "}")
+  defp interpolate_doc(doc), do: surround("\#{", doc, "}") |> format_to_string() |> string()
```

Comment:
Surround is the done adding the breaks. Instead of handling force, try this:

```suggestion
  defp interpolate_doc(doc), do: concat(["\#{", doc, "}"]) |> format_to_string() |> string()
```

Reply:
Also, please add a test where you make sure something like:

```elixir
"""
foo
#{concat(foo, bar)}
bar
"""
```

doesn't break lines either.

Reply:
When using `concat` directly instead of `surround` and handling `:doc_force`  the following test fails
```
test "with comment inside before and after" do
      assert_same ~S"""
      IO.puts(
        "Hello #{
          # comment
          world
        }"
      )
      """

      assert_same ~S"""
      IO.puts(
        "Hello #{
          world
          # comment
        }"
      )
      """
end
```

with the following message
```
1) test interpolation with comment inside before and after (Code.Formatter.CommentsTest)
     test/elixir/code_formatter/comments_test.exs:243
     Assertion with == failed
     code:  assert IO.iodata_to_binary(Code.format_string!(good, opts)) == String.trim(good)
     left:  "IO.puts(\"Hello \#{# comment\nworld}\")"
     right: "IO.puts(\n  \"Hello \#{\n    # comment\n    world\n  }\"\n)"
     stacktrace:
       test/elixir/code_formatter/comments_test.exs:244: (test)
```

This is similar when using `surround` as well. My thought was that this test should continue to pass with these changes because we want to keep the formatting as is when comments are in the interpolation

Is my understanding incorrect? Do we want this test to be updated as well?